--- a/finalforeach/cosmicreach/entities/player/PlayerEntity.java
+++ b/finalforeach/cosmicreach/entities/player/PlayerEntity.java
@@ -86,7 +_,7 @@
    public void onAttackInteraction(Entity sourceEntity) {
       if (sourceEntity instanceof PlayerEntity) {
          if (!ServerSettings.PVP_ENABLED.isFalse()) {
-            this.hit(1.0F);
+            this.hit(1.0F, this); // VoidCR - provide source entity
          }
       } else {
          super.onAttackInteraction(sourceEntity);
