--- a/finalforeach/cosmicreach/util/logging/Logger.java
+++ b/finalforeach/cosmicreach/util/logging/Logger.java
@@ -35,20 +_,24 @@
    }
 
    public static void info(Object message) {
-      log(Logger.Level.INFO, message.toString());
+      sh.miles.voidcr.adjust.redirect.NaiveLogger.info(message); // VoidCR - better logging
+      // log(Logger.Level.INFO, message.toString());
    }
 
    public static void warn(Object message) {
-      log(Logger.Level.WARNING, message.toString());
+      sh.miles.voidcr.adjust.redirect.NaiveLogger.warn(message); // VoidCR - better logging
+      // log(Logger.Level.WARNING, message.toString());
    }
 
    public static void error(Object message) {
-      log(Logger.Level.ERROR, message.toString());
+      sh.miles.voidcr.adjust.redirect.NaiveLogger.error(message); // VoidCR - better logging
+      // log(Logger.Level.ERROR, message.toString());
    }
 
    public static void debug(Object message) {
       if (debug) {
-         log(Logger.Level.DEBUG, message.toString());
+         sh.miles.voidcr.adjust.redirect.NaiveLogger.debug(message); // VoidCR - better logging
+         // log(Logger.Level.DEBUG, message.toString());
       }
    }
 
