--- a/finalforeach/cosmicreach/blocks/BlockState.java
+++ b/finalforeach/cosmicreach/blocks/BlockState.java
@@ -423,7 +_,7 @@
       if (this.stateGenerators != null) {
          for (String genKey : this.stateGenerators) {
             BlockStateGenerator generator = BlockStateGenerator.getInstance(genKey);
-            System.out.println("Applying generator " + genKey + " for blockstate: " + this.getSaveKey());
+            sh.miles.voidcr.adjust.redirect.NaiveLogger.info("Applying generator " + genKey + " for blockstate: " + this.getSaveKey()); // VoidCR - better logging
             generator.generate(this);
          }
       }
