--- a/finalforeach/cosmicreach/networking/netty/NettyServer.java
+++ b/finalforeach/cosmicreach/networking/netty/NettyServer.java
@@ -35,6 +_,7 @@
 import java.io.InputStreamReader;
 
 public class NettyServer {
+   public boolean running = true; // VoidCR - allow control of server termination
    public final ServerBroadcastIdentity broadcastIdentity;
    ObjectMap<ChannelHandlerContext, ServerIdentity> contextToIdentity = new ObjectMap();
    private Array<ServerIdentity> unauthenticatedConnections = new Array();
@@ -47,7 +_,7 @@
          if (account != null) {
             Logger.info(account.getUsername() + ": " + messageText);
          } else {
-            Logger.info("\n" + messageText);
+            Logger.info(messageText); // VoidCR - better logging
          }
       }
    };
@@ -90,11 +_,11 @@
          String inputString = null;
          Logger.info("Server started, type 'stop' to shutdown and 'help' for commands.");
 
-         while (!"stop".equals(inputString) && this.channelFuture.channel().isActive()) {
+         while (!"".equals(inputString) && this.channelFuture.channel().isActive() && running) { // VoidCR - allow control of server termination
             inputString = bufferRead.readLine();
             this.processCommand(inputString);
          }
-
+         sh.miles.voidcr.impl.server.VoidServer.SERVER.unload(); // VoidCR - unload server
          this.shutdown();
          this.channelFuture.channel().closeFuture().sync();
       } finally {
