--- a/finalforeach/cosmicreach/networking/netty/NettyServer.java
+++ b/finalforeach/cosmicreach/networking/netty/NettyServer.java
@@ -47,7 +_,7 @@
          if (account != null) {
             Logger.info(account.getUsername() + ": " + messageText);
          } else {
-            Logger.info("\n" + messageText);
+            Logger.info(messageText); // VoidCR - better logging
          }
       }
    };
@@ -86,6 +_,7 @@
                .option(ChannelOption.SO_BACKLOG, 128))
             .childOption(ChannelOption.SO_KEEPALIVE, true);
          this.channelFuture = b.bind(this.port).sync();
+         /* VoidCR start - smart console
          BufferedReader bufferRead = new BufferedReader(new InputStreamReader(System.in));
          String inputString = null;
          Logger.info("Server started, type 'stop' to shutdown and 'help' for commands.");
@@ -95,6 +_,13 @@
             this.processCommand(inputString);
          }
 
+          */
+         final var console = new sh.miles.voidcr.adjust.rewrite.SmartConsole();
+         console.spin();
+         while(ServerSingletons.running.get());
+         console.halt();
+         sh.miles.voidcr.impl.server.VoidServer.SERVER.unload(); // VoidCR - unload server
+         // VoidCR end - smart console
          this.shutdown();
          this.channelFuture.channel().closeFuture().sync();
       } finally {
