--- a/finalforeach/cosmicreach/world/World.java
+++ b/finalforeach/cosmicreach/world/World.java
@@ -20,6 +_,9 @@
 import sh.miles.voidcr.impl.world.VoidUniverse;
 
 public class World implements Serializable {
+   // VoidCR start
+   public final java.util.Map<String, Player> nameToPlayer = new HashMap<>(); // name to player for fast player resolution
+   // VoidCR end
    public String defaultZoneId;
    private transient HashMap<String, Zone> zoneMap = new HashMap<String, Zone>();
    public transient String worldFolderName;
