--- a/finalforeach/cosmicreach/blockentities/BlockEntity.java
+++ b/finalforeach/cosmicreach/blockentities/BlockEntity.java
@@ -14,6 +_,9 @@
 import java.util.HashMap;
 
 public abstract class BlockEntity implements IBlockEntity, ICRBinSerializable, ISlotContainerParent {
+   // VoidCR - apply generic API getter
+   public abstract sh.miles.voidcr.impl.world.block.entity.VoidBlockEntity<?> getVoidMirror();
+   // VoidCR - apply generic API getter
    public transient HashMap<String, Runnable> signalMap;
    Zone zone;
    int x;
