--- /dev/null
+++ b/finalforeach/cosmicreach/settings/BooleanSetting.java
@@ -1,0 +_,43 @@
+package finalforeach.cosmicreach.settings;
+
+public class BooleanSetting extends GameSetting {
+   private final String key;
+   private boolean value;
+
+   public BooleanSetting(String key, boolean defaultValue) {
+      this.key = key;
+      Object mapping = ALL_SETTINGS.getOrDefault(key, defaultValue);
+      if (mapping instanceof Boolean) {
+         this.value = (Boolean)mapping;
+      } else {
+         this.value = defaultValue;
+      }
+   }
+
+   public boolean getValue() {
+      return this.value;
+   }
+
+   public boolean isTrue() {
+      return this.getValue();
+   }
+
+   public boolean isFalse() {
+      return !this.getValue();
+   }
+
+   private void save() {
+      ALL_SETTINGS.put(this.key, this.value);
+      saveSettings();
+   }
+
+   public void setValue(boolean newValue) {
+      this.value = newValue;
+      this.save();
+   }
+
+   public void toggleValue() {
+      this.value = !this.value;
+      this.save();
+   }
+}
