--- /dev/null
+++ b/finalforeach/cosmicreach/settings/CharSetting.java
@@ -1,0 +_,34 @@
+package finalforeach.cosmicreach.settings;
+
+public class CharSetting extends GameSetting {
+   private final String key;
+   private char value;
+
+   public CharSetting(String key, char defaultValue) {
+      this.key = key;
+      if (ALL_SETTINGS.getOrDefault(key, defaultValue) instanceof Character c) {
+         this.value = c;
+      } else {
+         this.value = defaultValue;
+      }
+   }
+
+   public char getValue() {
+      return this.value;
+   }
+
+   private void save() {
+      if (this.value == 0) {
+         ALL_SETTINGS.remove(this.key);
+      } else {
+         ALL_SETTINGS.put(this.key, this.value);
+      }
+
+      saveSettings();
+   }
+
+   public void setValue(char newValue) {
+      this.value = newValue;
+      this.save();
+   }
+}
