--- a/finalforeach/cosmicreach/server/ServerLauncher.java
+++ b/finalforeach/cosmicreach/server/ServerLauncher.java
@@ -32,7 +_,7 @@
                new File(SaveLocation.saveLocationOverride).mkdirs();
             }
          }
-
+         sh.miles.voidcr.impl.server.VoidServer.SERVER.getPluginLoader().loadPlugins(); // VoidCR - load plugins
          ServerSingletons.create();
          String var4 = "server-world-1";
          GameSingletons.world = ChunkLoader.loadWorld(var4);
