--- a/finalforeach/cosmicreach/savelib/crbin/CRBinSerializer.java
+++ b/finalforeach/cosmicreach/savelib/crbin/CRBinSerializer.java
@@ -225,7 +_,7 @@
       if (var4 != null) {
          var4.accept(this, var2, var3);
       } else if (ICRBinSerializable.class.isAssignableFrom(var1)) {
-         this.writeObj(var2, (T)((ICRBinSerializable)var3));
+         this.writeObj(name, ((ICRBinSerializable)item)); // VoidCR decompile fix
       } else {
          throw new RuntimeException(
             var1.getSimpleName()
