--- a/finalforeach/cosmicreach/savelib/crbin/CRBinSerializer.java
+++ b/finalforeach/cosmicreach/savelib/crbin/CRBinSerializer.java
@@ -228,7 +_,7 @@
       if (consumer != null) {
          consumer.accept(this, name, (T)item);
       } else if (ICRBinSerializable.class.isAssignableFrom(elementType)) {
-         this.writeObj(name, (T)((ICRBinSerializable)item));
+         this.writeObj(name, ((ICRBinSerializable)item)); // VoidCR decompile fix
       } else {
          throw new RuntimeException(
             elementType.getSimpleName()
