--- a/finalforeach/cosmicreach/savelib/crbin/CRBinDeserializer.java
+++ b/finalforeach/cosmicreach/savelib/crbin/CRBinDeserializer.java
@@ -285,7 +_,7 @@
          if (var4 == null) {
             return null;
          } else if (ICRBinSerializable.class.isAssignableFrom(var2)) {
-            return this.<T>readObj(var2, var4);
+            return (T) this.readObj((Class<? extends ICRBinSerializable>) elementType, d); // VoidCR - decompile
          } else {
             throw new RuntimeException(
                var2.getSimpleName()
