--- a/finalforeach/cosmicreach/ZoneLoaders.java
+++ b/finalforeach/cosmicreach/ZoneLoaders.java
@@ -51,7 +_,7 @@
             }
          }
 
-         if (GameSingletons.getNumberOfPlayers() == 0 && !GameSingletons.isClient || this.zoneLoaders.isEmpty()) {
+         if (GameSingletons.getNumberOfPlayers() == 0 && finalforeach.cosmicreach.networking.server.ServerSingletons.running.get() && !GameSingletons.isClient || this.zoneLoaders.isEmpty()) {
             synchronized (this.worldGenThread) {
                if (GameSingletons.getNumberOfPlayers() == 0 && !GameSingletons.isClient || this.zoneLoaders.isEmpty()) {
                   Entries var8 = this.zoneLoaders.entries().iterator();
