--- a/finalforeach/cosmicreach/entities/DroneTrapEntity.java
+++ b/finalforeach/cosmicreach/entities/DroneTrapEntity.java
@@ -103,7 +_,7 @@
    @Override
    public void onUseInteraction(Player player, ItemStack heldItemStack) {
       super.onUseInteraction(player, heldItemStack);
-      if (this.friendly) {
+      if (this.friendly || sh.miles.voidcr.impl.server.VoidServer.SERVER.getConfiguration().behaviorChanges().activateDroneTrapOnRightClick(player, heldItemStack)) { // VoidCR - allow release of drone trap in creative
          this.activateTrap();
       }
    }
@@ -154,7 +_,7 @@
 
       EntityUtils.nudgeEntityFromOthers(this);
       if (!this.friendly) {
-         Entity closestPlayerEntity = EntityUtils.getClosestPlayerToEntity(this, zone);
+         Entity closestPlayerEntity = EntityUtils.getClosestPlayerToEntity(this, zone, (p) -> p.gamemode != finalforeach.cosmicreach.entities.player.Gamemode.CREATIVE); // VoidCR - Don't target creative players
          if (closestPlayerEntity != null && this.timeSinceActivated < 0.0F) {
             this.activateTrap();
          }
