--- a/finalforeach/cosmicreach/entities/DroneEntity.java
+++ b/finalforeach/cosmicreach/entities/DroneEntity.java
@@ -171,7 +_,7 @@
          this.tmpDesiredLocation.add(0.0F, hoverHeight, 0.0F);
          this.tmpDesiredLocation.add(targetEntity.getPosition());
          if (!this.isTargetingFriend && targetEntity.globalBoundingBox.intersects(this.globalBoundingBox)) {
-            targetEntity.hit(1.0F);
+            targetEntity.hit(1.0F, this); // VoidCR - provide source entity
             this.attackChargeTime = MathUtils.random(20, 60);
          }
       }
@@ -220,6 +_,12 @@
       if (target == null) {
          target = EntityUtils.getClosestPlayerToEntity(this, zone);
       }
+
+      // VoidCR start - don't target dead entities
+      if (target.isDead()) {
+         target = null;
+      }
+      // VoidCR end - don't target dead entities
 
       return target;
    }
