--- a/finalforeach/cosmicreach/entities/Entity.java
+++ b/finalforeach/cosmicreach/entities/Entity.java
@@ -45,6 +_,9 @@
    private static final BlockPosition tmpBlockPos1 = new BlockPosition(null, 0, 0, 0);
    private static final BlockPosition tmpBlockPos2 = new BlockPosition(null, 0, 0, 0);
    private static EntityPositionPacket positionPacket = new EntityPositionPacket();
+   // VoidCR start - Extra Fields
+   public int maxInvulnerabilityFrames = 5;
+   // VoidCR end - Extra Fields
    public boolean shouldSaveInChunks = true;
    @CRBSerialized
    public EntityUniqueId uniqueId = EntityUniqueId.getNew();
@@ -279,19 +_,29 @@
    public boolean hasTag(String tag) {
       return this.tags == null ? false : this.tags.contains(tag, false);
    }
-
+   // VoidCR start - add method parameter to carry through sourceEntity
    public void hit(float amount) {
+      hit(amount, null);
+   }
+   public void hit(float amount, Entity sourceEntity) {
       if (amount != 0.0F) {
+         if (sourceEntity != null && sourceEntity.isDead()) return; // fix event trigger while dead
+         final var event = sh.miles.voidcr.impl.plugin.lifecycle.event.VoidEventFactory.preEntityDamageEvent(amount, maxInvulnerabilityFrames, sourceEntity, this);
+         amount = event == null ? amount : event.getDamage();
+         // VoidCR end - add method parameter to carry through sourceEntity
+
          if (ServerSingletons.SERVER != null) {
             ServerSingletons.SERVER.broadcast(this.zone, new HitEntityPacket(this, amount));
          }
-
          if (amount < 0.0F) {
             this.hitpoints = Math.min(this.hitpoints - amount, this.maxHitpoints);
          } else if (this.getPendingDamage() == 0.0F && this.invulnerabiltyFrames <= 0) {
+            // VoidCR start - PreEntityDamageEvent
             this.setPendingDamage(Math.max(this.getPendingDamage(), amount));
-            this.invulnerabiltyFrames = 5;
+            this.invulnerabiltyFrames = event == null ? maxInvulnerabilityFrames : event.getInvulnerabilityFrames(); // apply max invulnerability frames
+            // VoidCR end - PreEntityDamageEvent
          }
+         sh.miles.voidcr.impl.plugin.lifecycle.event.VoidEventFactory.postEntityDamageEvent(this.pendingDamage, this.invulnerabiltyFrames, sourceEntity, this); // VoidCR - PostEntityDamageEvent
       }
    }
 
@@ -318,7 +_,7 @@
    }
 
    public void onAttackInteraction(Entity sourceEntity) {
-      this.hit(100.0F);
+      this.hit(100.0F, sourceEntity); // VoidCR provide sourceEntity
    }
 
    public void onCollideWithBlock(Axis axis, BlockState block, int bx, int by, int bz) {
