--- a/finalforeach/cosmicreach/entities/EntityUtils.java
+++ b/finalforeach/cosmicreach/entities/EntityUtils.java
@@ -42,7 +_,7 @@
          Player p = (Player)players.get(i);
          if (p != null && (canTarget == null || canTarget.test(p))) {
             Entity pe = p.getEntity();
-            if (pe != null) {
+            if (pe != null && !pe.isDead()) { // VoidCR - don't target dead players
                float dst = sourceEntity.position.dst(pe.position);
                if (!(dst > range)) {
                   if (closest == null) {
