--- a/finalforeach/cosmicreach/entities/Entity.java
+++ b/finalforeach/cosmicreach/entities/Entity.java
@@ -34,7 +_,11 @@
 import java.util.function.Consumer;
 import java.util.function.Predicate;
 
-public class Entity implements ICRBinSerializable {
+public class Entity implements ICRBinSerializable {// VoidCR start - add API conversion
+   private sh.miles.voidcr.impl.entity.VoidEntity cache;
+   protected sh.miles.voidcr.impl.entity.VoidEntity createVoidEntity() { return new sh.miles.voidcr.impl.entity.VoidEntity(this); }
+   public sh.miles.voidcr.impl.entity.VoidEntity getVoidMirror() { return cache == null ? cache = createVoidEntity() : cache; }
+   // VoidCR end - add API conversion
    public static Vector3 gravity = new Vector3(0.0F, -29.400002F, 0.0F);
    public boolean shouldSaveInChunks = true;
    @CRBSerialized
